yt_dlp/extractor/rtp.py

```diff
@@ -25,42 +26,87 @@ class RTPIE(InfoExtractor):
         'only_matching': True,
     }]
 
+    _RX_OBFUSCATION = re.compile(r'''(?xs)
+        atob\s*\(\s*decodeURIComponent\s*\(\s*
+            (\[[0-9A-Za-z%,'"]*\])
+        \s*\.\s*join\(\s*(?:""|'')\s*\)\s*\)\s*\)
+    ''')
+
+    def __unobfuscate(self, data, *, video_id):
```

Comment:
since there aren't many fields and all are mandatory, I don't really see the need for `*`. I'm not bothered by it enough to ask it to be removed, but I'm curious as to your reason for using it

Reply:
I worry a little that more such extra context parameters may proliferate, quickly making things unreadable.

I’d have preferred not to pass it in the first place, honestly. What does `_parse_json` need it for anyway?